api/lightwallet_rest.md

```diff
@@ -0,0 +1,402 @@
+> If the total received outputs for the address is less than `amount`, the
+> server shall return a HTTP 400 "Bad Request" error code.
+
+**Response** object
+
+|    Field     |        Type       |                Description              |
+|--------------|-------------------|-----------------------------------------|
+| per_byte_fee | uint64-string     | Estimated network fee                   |
+| fee_mask     | uint64-string     | Fee quantization mask                   |
+| amount       | uint64-string     | The total value in outputs              |
+| outputs      | array of output's | Outputs possibly available for spending |
+
+#### `import_request`
+Request an account scan from the genesis block.
+
+**Request** object
+
+|   Field  |      Type      |      Description        |
+|----------|----------------|-------------------------|
+| address  | base58-address | Address to create/probe |
+| view_key | binary         | View key bytes          |
+
+**Response** object
+
+|       Field        |      Type      |           Description            |
+|--------------------|----------------|----------------------------------|
+| payment_address *  | base58-address | Payment location                 |
+| payment_id *       | binary         | Bytes for payment_id tx field    |
+| import_fee *       | uint64-string  | Fee required to complete request |
+| new_request        | boolean        | New or existing request          |
+| request_fulfilled  | boolean        | Indicates success                |
+| status             | string         | Custom message                   |
+
+> `payment_id`, `import_fee`, and `payment_address` may be omitted if the
+> client does not need to send XMR to complete the request.
+
+#### `login`
+Check for the existence of an account or create a new one.
+
+**Request** object
+
+|       Field       |      Type      |           Description            |
+|-------------------|----------------|----------------------------------|
+| address           | base58-address | Address to create/probe          |
+| view_key          | binary         | View key bytes                   |
+| create_account    | boolean        | Try to create new account        |
+| generated_locally | boolean        | Indicate that the address is new |
+
+> The view key bytes are required even if an account is not being created, to
+> prevent metadata leakage.
+
+> If the server does not allow account creations, HTTP 501 "Not Implemented"
+> error must be returned.
+
+> If approval process is manual, a successful HTTP 200 OK and response object
+> must be returned. Subsequent requests shall be HTTP 405 "Forbidden" until
+> account is approved.
+
+**Response** object
+
+|        Field        |  Type   |            Description             |
+|---------------------|---------|------------------------------------|
+| new_address         | boolean | Whether account was just created   |
+| generated_locally * | boolean | Flag from initial account creation |
```

Comment:
What does `*` mean in this context?

Reply:
What is the purpose of having a response field as optional?

Reply:
It is not required for the server to implement the feature. Not sure if this is the best approach, but that was the idea (this was added very recently to mymonero and openmonero has (had?) no such concept).